code/espurna/sensors/VL53L1XSensor.h

```diff
@@ -0,0 +1,108 @@
+// -----------------------------------------------------------------------------
+// VL53L1X Sensor over I2C
+// Copyright (C) 2017-2018 by Xose Pérez <xose dot perez at gmail dot com>
+// -----------------------------------------------------------------------------
+
+#if SENSOR_SUPPORT && VL53L1X_SUPPORT
+
+#pragma once
+
+#undef I2C_SUPPORT
+#define I2C_SUPPORT 1 // Explicitly request I2C support.
+
+
+#include "Arduino.h"
+#include "I2CSensor.h"
+#include "VL53L1X.h"
+
+class VL53L1XSensor : public I2CSensor {
+
+    public:
+
+        // ---------------------------------------------------------------------
+        // Public
+        // ---------------------------------------------------------------------
+
+        VL53L1XSensor(): I2CSensor() {
+            _count = 1;
+            _sensor_id = SENSOR_VL53L1X_ID;
+            _vl53l1x = new VL53L1X();
+        }
+
+        ~VL53L1XSensor() {
+          delete _vl53l1x;
+        }
+
+        // ---------------------------------------------------------------------
+
+        void setDistanceMode(VL53L1X::DistanceMode mode) {
+          _vl53l1x->setDistanceMode(mode);
+        }
+
+        void setMeasurementTimingBudget(uint32_t budget_us) {
+          _vl53l1x->setMeasurementTimingBudget(budget_us);
+        }
+
+        void setInterMeasurementPeriod(unsigned int period) {
+          if (_inter_measurement_period == period) return;
+          _inter_measurement_period = period;
+          _dirty = true;
+        }
+
+        // ---------------------------------------------------------------------
+        // Sensor API
+        // ---------------------------------------------------------------------
+
+        void begin() {
+          if (!_dirty) {
```

Comment:
I2C sensors can benefit from address auto-discover feature. Check the BMX280 for instance.

Reply:
Will take a look. Mind if I add this as a separate PR?